src/plugins/discover/public/application/apps/main/components/chart/discover_chart.tsx

```diff
@@ -8,45 +8,43 @@
 import React, { useCallback, useEffect, useRef } from 'react';
 import moment from 'moment';
 import { EuiFlexGroup, EuiFlexItem, EuiButtonEmpty, EuiSpacer } from '@elastic/eui';
-import { IUiSettingsClient } from 'kibana/public';
 import { i18n } from '@kbn/i18n';
 import { HitsCounter } from '../hits_counter';
-import { DataPublicPluginStart, IndexPattern, search } from '../../../../../../../data/public';
+import { IndexPattern, search } from '../../../../../../../data/public';
 import { TimechartHeader } from '../timechart_header';
 import { SavedSearch } from '../../../../../saved_searches';
 import { AppState, GetStateReturn } from '../../services/discover_state';
-import { TimechartBucketInterval } from '../timechart_header/timechart_header';
 import { Chart as IChart } from './point_series';
 import { DiscoverHistogram } from './histogram';
+import { DataMain$ } from '../../services/use_saved_search';
+import { DiscoverServices } from '../../../../../build_services';
 
 const TimechartHeaderMemoized = React.memo(TimechartHeader);
 const DiscoverHistogramMemoized = React.memo(DiscoverHistogram);
 export function DiscoverChart({
-  config,
-  data,
-  bucketInterval,
-  chartData,
-  hits,
   isLegacy,
   resetQuery,
   savedSearch,
+  savedSearchDataChart$,
+  savedSearchDataTotalHits$,
+  services,
   state,
   stateContainer,
   timefield,
 }: {
-  config: IUiSettingsClient;
-  data: DataPublicPluginStart;
-  bucketInterval?: TimechartBucketInterval;
   chartData?: IChart;
-  hits?: number;
   indexPattern: IndexPattern;
   isLegacy: boolean;
   resetQuery: () => void;
   savedSearch: SavedSearch;
+  savedSearchDataChart$: DataMain$;
+  savedSearchDataTotalHits$: DataMain$;
```

Comment:
Shouldn't that be `DataTotalHits$` and `DataCharts$` type?

Reply:
true, thx!